core/src/test/java/nl/nn/adapterframework/jdbc/FixedQuerySenderTest.java

```diff
@@ -72,71 +73,166 @@ public void setUp() throws Exception {
 	}
+
+	@Test
+	public void testNamedParameters() throws Exception {
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', ?{param})");
+		Parameter param = new Parameter();
+		param.setName("param");
+		param.setValue("value");
+		sender.addParameter(param);
+
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("<result><rowsupdated>1</rowsupdated></result>", result.asString());
+	}
+
+	@Test
+	public void testUseNamedParametersStringValueContains_unp_start() throws Exception {
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('3', '?{param}')");
+
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("<result><rowsupdated>1</rowsupdated></result>", result.asString());
+		
+		sender.setQuery("SELECT TVARCHAR FROM TEMP WHERE TKEY='3'");
+		sender.setQueryType("select");
+		sender.setScalar(true);
+
+		result = sendMessage("dummy");
+
+		assertEquals("?{param}", result.asString());
+	}
+
+	@Test
+	public void testUseNamedParametersStringValueContains_unp_start_resolveParam() throws Exception {
+		exception.expect(SenderException.class);
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', '?{param}')");
+
+		Parameter param = new Parameter();
+		param.setName("param");
+		param.setValue("value");
+		sender.addParameter(param);
+
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("?{param}", result.asString());
+	}
+	
+	@Test
+	public void testUseNamedParametersWithoutNamedParam() throws Exception {
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', 'text')");
+		sender.setUseNamedParams(true);
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("<result><rowsupdated>1</rowsupdated></result>", result.asString());
+	}
+
+	@Test
+	public void testUseNamedParametersWithoutParam() throws Exception {
+		exception.expect(SenderException.class);
+		exception.expectMessage("Syntax error in SQL statement \"INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', ?{param})[*]\"");
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', ?{param})");
+		sender.setUseNamedParams(true);
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("<result><rowsupdated>1</rowsupdated></result>", result.asString());
+	}
 	
+	@Test
+	public void testNamedParamInQueryFlagFalse() throws Exception {
+		exception.expect(SenderException.class);
+		exception.expectMessage("Syntax error in SQL statement \"INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', ?{param})[*]\"");
+		sender.setQuery("INSERT INTO TEMP (TKEY, TVARCHAR) VALUES ('1', ?{param})");
+		sender.setUseNamedParams(false);
+		sender.configure();
+		sender.open();
+
+		Message result = sendMessage("dummy");
+		assertEquals("<result><rowsupdated>1</rowsupdated></result>", result.asString());
```

Comment:
Could you remove this assertEquals and move the exception.expect to just before the sendMessage()? 
As it is now it is confusing if you don't read very carefully.